### ISSUE TITLE
[RUNTIME] TPU Backend Support

### ISSUE THREAD

**Issue by merrymercy** [alpa-projects/alpa#433]
# Background
Currently, Alpa only supports GPU. With XLA as Alpa's backend, it is easy to support TPU as well. We can call the auto-sharding pass to generate the sharding annotated HLO and then call the closed-source TPU compiler.

# TODO
  - [ ] Support shard_parallel on TPU
  - [ ] Support pipeshard_parallel on TPU

**Comment by OhadRubin** [comment]
What is the status of this issue?
It would be nice to run a larger models on multiple v3-8.

**Comment by merrymercy** [comment]
@OhadRubin It should be supported very soon. See our [tpu-support](https://github.com/alpa-projects/alpa/commits/tpu-support) branch and the official tensorflow [code](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/compiler/xla/experimental/auto_sharding)

**Comment by merrymercy** [comment]
#764

**Comment by zhisbug** [comment]
Is TPU+pipeshard still in our scope? @merrymercy @ZYHowell

**Comment by merrymercy** [comment]
No

**Comment by jon-chuang** [comment]
Hello. TPU support branch does not exist. Could I ask if this is still being worked on?

**Comment by Lime-Cakes** [comment]
Pipeshard isn't going to be supported on TPU?

**Comment by ZYHowell** [comment]
We have no plan mainly because 1) the TPU backend of XLA is close sourced; 2) unlike NCCL for GPU, the TPU has no communication library exposed

**Comment by Lime-Cakes** [comment]
Thanks!

Is  there going to be none-cuda jaxlib (the alpa fork)? And is there some guide for installing alpa for TPU? It seems all the guide is for CUDA machines.

**Comment by ZYHowell** [comment]
Yes I think so. But in case we only supports shard parallel in the TPU side, and the part is already in the upstream official jax/jaxlib support, I'd suppose you to try the `pjit.auto`
